OpenSim/Simulation/Test/testForces.cpp

```diff
@@ -848,12 +848,35 @@ void testFunctionBasedBushingForce()
         ASSERT_EQUAL(analytical_force, model_force[7], 2e-4);
     }
 
-    osimModel.disownAllComponents();
-
     manager.getStateStorage().print("function_based_bushing_model_states.sto");
 
     // Save the forces
     reporter->getForceStorage().print("function_based_bushing_forces.mot");  
+    // Now add damping to the bushing force and make sure energy is monotonically decreasing
+    spring.set_translational_damping(Vec3(100.0));
+    SimTK::State& osim_state2 = osimModel.initSystem();
+
+    // set the initial height of the ball on slider
+    sliderCoord.setValue(osim_state, start_h);
+    osimModel.getMultibodySystem().realize(osim_state2, Stage::Position);
+
+    //==========================================================================
+    // Compute the Energy to make sure it goes down due to damping
+    Manager manager2(osimModel);
+    manager2.setIntegratorAccuracy(1e-6);
+    osim_state2.setTime(0.0);
+    manager2.initialize(osim_state2);
+
+    double lastEnergy = 1E20;   // Large
+    for (int i = 1; i <= nsteps; ++i) {
+        osim_state2 = manager2.integrate(dt*i);
+        osimModel.getMultibodySystem().realize(osim_state2, Stage::Acceleration);
+        double newEnergy = osimModel.calcKineticEnergy(osim_state2) + osimModel.calcPotentialEnergy(osim_state2);
+        ASSERT(newEnergy < lastEnergy);
+        lastEnergy = newEnergy;
+    }
+
+    osimModel.disownAllComponents();
```

Comment:
This comment from 4 years ago is relevant (and note that my code was just going into the sandbox): https://github.com/opensim-org/opensim-core/pull/663#discussion_r41592947

Reply:
True @tkuchida, though I didn't introduce this line I just moved it to reuse the model. I can add a comment stating this is bad if you think it would be helpful. We could use heap allocated objects and the line would not be needed but the code will look more ugly.

Reply:
I don't have a strong opinion (though others may). There's already an Issue about disowning the `disownAllComponents()` method (#148) which seems fine to me.

Reply:
I added a comment as  suggested